cmd/snap-confine/snap-confine.apparmor.in

```diff
@@ -396,4 +396,7 @@
 
     # Allow snap-confine to be killed
     signal (receive) peer=unconfined,
+
+    # Allow snap-confine to use snap-update-ns
+    /{var/lib/snapd/hostfs/,}{,{,var/lib/snapd/}snap/core/*/}usr/lib{,exec}/snapd/snap-update-ns Ux,
```

Comment:
I'm not clear about syntax of apparmor profiles here: it's clear `/{var/lib/snapd/hostfs/,}` is an obsolute path, but how about remaining paths?

Reply:
This rule is not clear. Can you enumerate the paths? Maybe we could then split the rule into two rules.

Reply:
This rule is a composition of the following possibilities:

- /usr/lib/snapd/snap-update-ns
- /usr/libexec/snapd/snap-update-ns

- same as above but when using core snap mounted at /snap/core/*/
- same as above but when using core snap mounted at /var/lib/snapd/snap/core/*
- same as above but when all of those are visible through hostfs at /var/lib/snapd/hostfs

Reply:
From IRC, I think we should break this up a little to make it easier to audit:
```
// Allow executing snap-update-ns when:
// not re-execing
/usr/lib{,exec}/snapd/snap-update-ns Ux,
// when re-execing
/snap/core/*/usr/lib{,exec}/snapd/snap-update-ns Ux,
// when ???
/var/lib/snapd/{hostfs,snap/core/*}/usr/lib{,exec}/snapd-update-ns Ux,
```

Reply:
Note, if within the core snap, do we ever need `lib{,exec}`?

Reply:
Note they aren't regular expressions. They are AARE which is much more limited (see man apparmor.d for details).